### PR TITLE
feat(sdk): SDK - Components - Added support for editing components

### PR DESCRIPTION
**Pull request by Ark-kun**
Added method that generates code to edit the component YAML text. This simplifies editing components that are loaded from local or remote files.

Example:

```python
        my_op = components.load_component_from_url(...)
        my_op.edit()
```

### CODE REVIEW

**Comment by Bobgy** [comment]
/assign @numerology 
for review

**Comment by Ark-kun** [comment]
@numerology Can you please take a look?

File: sdk/python/kfp/components/_editing.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    This function can work with components loaded from URLs, files or text.
+    This function is used indirectly by calling `op.edit()` method.
+    If used in a notebook, adds cell with the component editing code.
+    Otherwise returns the code as string.
+
+    Example::
+
+        my_op = components.load_component_from_*(...)
+        my_op.edit()
```
Comment (numerology) [inline sdk/python/kfp/components/_editing.py:33]: Maybe add two more lines briefly showing how to edit fields like image/args?

File: sdk/python/kfp/components/_editing.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    editing_code = generate_code_to_edit_component_yaml(component_ref=component_ref)
+    # If we're in a notebook, add a new cell. Otherwise return the editing code as string
+    try:
+        _ipython_create_new_cell(editing_code)
+    except:
+        return editing_code
+
+
+def generate_code_to_edit_component_yaml(
+    component_ref: ComponentReference, component_store: ComponentStore = None,
```
Comment (numerology) [inline sdk/python/kfp/components/_editing.py:44]:
```suggestion
    component_ref: ComponentReference, component_store: Optional[ComponentStore] = None,
```

File: sdk/python/kfp/components/_editing.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    if not component_spec:
+        if not component_store:
+            component_store = ComponentStore.default_store
+        component_ref = component_store._load_component_spec_in_component_ref(component_ref)
+        component_spec = component_ref.spec
+
+    component_data = component_spec._data
+    local_path = getattr(component_spec, '_local_path', None)
+    if not local_path:
+        if component_ref.url:
```
Comment (numerology) [inline sdk/python/kfp/components/_editing.py:60]: I think we should at least add some logging message here for this implicit side effect, to inform users that compiler is writing out to a local file.

File: sdk/python/kfp/components_tests/test_components.py [hunk 4]
```diff
@@ -693,6 +699,64 @@ def test_check_task_object_no_output_attribute_when_multiple_outputs(self):
 
         self.assertFalse(hasattr(task, 'output'))
 
+    def test_edit_component_loaded_from_text(self):
```
Comment (numerology) [inline sdk/python/kfp/components_tests/test_components.py:702]: Can we try to add a test case where we actually mutate the spec of the component and assert the changes is respected? Thanks

**Review by numerology — COMMENTED** [review]
Thanks!

File: sdk/python/kfp/components_tests/test_components.py (hunk 4 above)
Comment (zijianjoy) [reply sdk/python/kfp/components_tests/test_components.py:702]: +1, I am interested to see how to programmatically edit a component, and how does it benefit users.

Comment (Ark-kun) [reply sdk/python/kfp/components_tests/test_components.py:702]: Programmatic editing is already possible:

```python
my_op = load_component_from_uri(...)

# Modify my_op.component_spec
my_op.component_spec.save("new.component.yaml")
```

This is not intended for programmatic editing, but rather for users of Jupyter notebooks: You call .edit() and the editable component spec is added as the next cell.

**Pull request closed without merging**